base/shell/explorer/traywnd.cpp

```diff
@@ -264,6 +264,95 @@ class CStartButton
 
 };
 
+static const WCHAR szTrayShowDesktopButton[] = L"TrayShowDesktopButtonWClass";
+
+// The 'Show Desktop' button at edge of taskbar
+class CTrayShowDesktopButton :
+    public CComCoClass<CTrayShowDesktopButton>,
+    public CComObjectRootEx<CComMultiThreadModelNoCS>,
```

Comment:
Are the `CComCoClass` and `CComObjectRootEx` required here?
I think you only need the `CWindowImpl` in this case?

Reply:
Removed `CComCoClass` and `CComObjectRootEx`.